### PR TITLE
#161255188  configure jest test coverage collection

### PR DESCRIPTION
**Pull request by p8ul**
#### What does this PR do?
- Configure the environment for writing tests
#### Description of Task to be completed?
- strive to increase test coverage
- configure jest collect coverage
#### How should this be manually tested?
 `yarn run coveralls`
#### What are the relevant pivotal tracker stories?
[#161255188](https://www.pivotaltracker.com/story/show/161255188)

### CODE REVIEW

**Review by MartinKaburu — APPROVED** [review]
I approve the changes, great work.

**Comment by Tony-Ndichu** [comment]
Looks good to me

**Review by Tony-Ndichu — APPROVED** [review]
Looks good to me

**Review by Mirriam-Maina — APPROVED** [review]
It is comprehensive and well organized. Looks great.

**Review by mmosoroohh — APPROVED** [review]
Recommended workflow...
Looks good.

File: src/routes/__tests__/routes.test.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import indexRoutes from '..';
+
+describe('Routes tests', () => {
+  it('should be an object of routs', () => {
+    expect(indexRoutes).toBeInstanceOf(Object);
```
Comment (Rodgers-M) [inline src/routes/__tests__/routes.test.js]: this does not really prove that it is a route object. can you assert the properties of the object?  have a look at jest `toHaveProperty` method

[link to method](https://jestjs.io/docs/en/expect#tohavepropertykeypath-value)

**Comment by Tony-Ndichu** [comment]
In src/routes/__tests__/routes.test.js line 4, I think you have a typo there, I think you meant "routes" instead of routs

File: src/routes/__tests__/routes.test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import indexRoutes from '..';
+
+
+describe('Routes tests', () => {
+  indexRoutes.map((value, index) => {
+    it('should be an object of routes', () => {
+      expect(indexRoutes[index]).toHaveProperty('path');
+      expect(indexRoutes[index]).toHaveProperty('component');
+    });
+    return true;
```
Comment (Rodgers-M) [inline src/routes/__tests__/routes.test.js]: why do you have a return value in tests?

File: src/routes/__tests__/routes.test.js (hunk 1 above)
Comment (p8ul) [reply src/routes/__tests__/routes.test.js]: Thanks for the review. The link was very helpful. I have updated the test using `toBeInstanceOf`.

File: src/routes/__tests__/routes.test.js (hunk 2 above)
Comment (p8ul) [reply src/routes/__tests__/routes.test.js]: Linting is suggesting that there should be a return. 
screen shot 2018-11-28 at 11 16 51
Am using map function to loop through all the route objects.

Comment (Rodgers-M) [reply src/routes/__tests__/routes.test.js]: you do not have to map over the index routes array, you can easily do 
`indexRoutes[0]` to access the object inside and assert for its properties.

Comment (p8ul) [reply src/routes/__tests__/routes.test.js]: I had done that, then I realized that in future, we will add more object routes in the array to take care of this, I found a workaround by using the map function to map through all the routes object and testing them individually. 

`{
    path: '/',
    name: 'Landing Page',
    component: LandingPage,
  },
  {
    path: '/login',
    name: 'Login Page',
    component: Login,
  },`

File: src/routes/__tests__/routes.test.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import indexRoutes from '..';
+
+
+describe('Routes tests', () => {
+  indexRoutes.map((value, index) => it('should be an object of routes', () => {
```
Comment (p8ul) [inline src/routes/__tests__/routes.test.js]: I have removed `return` statement from the map. Thank you for the feedback.

**Review by Rodgers-M — APPROVED** [review]

**Pull request merged**